WebService/v1/Controllers/SimulationsController.cs

```diff
@@ -92,6 +97,17 @@ public async Task<SimulationApiModel> GetAsync(string id)
                 simulation, this.servicesConfig, this.deploymentConfig, this.connectionStringManager, this.simulationRunner, this.rateReporter);
         }
 
+        [HttpPost("{id}/metrics/iothub")]
+        public async Task<object> PostAsync(
+            [FromBody] MetricsRequestsApiModel requests,
+            [FromHeader]string authorization,
+            string id)
+        {
+            return requests == null
+                ? await this.iothubMetrics.GetIothubMetrics(authorization, null)
+                : await this.iothubMetrics.GetIothubMetrics(authorization, requests.ToServiceModel());
```

Comment:
can you try this:

var requestsModel = requests?.ToServiceModel();
await this.iothubMetrics.GetIothubMetrics(authorization, requestsModel);